introspection-engine/introspection-engine-tests/tests/tables/postgres.rs

```diff
@@ -245,3 +245,36 @@ async fn a_table_with_json_columns(api: &TestApi) -> TestResult {
     api.expect_datamodel(&expected).await;
     Ok(())
 }
+
+#[test_connector(tags(Postgres), exclude(CockroachDb))]
+async fn datetime_default_expressions_are_not_truncated(api: &TestApi) -> TestResult {
+    let setup = r#"
+        CREATE TABLE "Foo" (
+            "id" INTEGER NOT NULL,
+            trial_expires TIMESTAMPTZ(6) NOT NULL DEFAULT now()::TIMESTAMPTZ + '14 days'::INTERVAL,
+
+            CONSTRAINT "Foo_pkey" PRIMARY KEY ("id")
+        );
+    "#;
+
+    api.raw_cmd(setup).await;
+
+    let expected = expect![[r#"
+        generator client {
+          provider = "prisma-client-js"
+        }
+
+        datasource db {
+          provider = "postgresql"
+          url      = "env(TEST_DATABASE_URL)"
+        }
+
+        model Foo {
+          id            Int      @id
+          trial_expires DateTime @default(dbgenerated("(now() + '14 days'::interval)")) @db.Timestamptz(6)
```

Comment:
Why the first cast gets removed, but not the second one?

Reply:
I don't know how the database decides that :shrug: